src/datatypes/scenario/FieldMap.cpp

```diff
@@ -49,6 +51,65 @@ namespace spy::scenario {
                  || map.at(p.getY()).size() <= static_cast<unsigned int>(p.getX()));
     }
 
+
+    /**
+     * @see playtechs.blogspot.com/2007/03/raytracing-on-grid.html for the original implementation
+     */
+    bool FieldMap::isLineOfSightFree(util::Point p1, util::Point p2) const {
+        if (!isInside(p1) || !isInside(p2)) {
+            throw std::invalid_argument("At least one point is outside the field map!");
+        } else if (p1 == p2) {
+            throw std::invalid_argument("Points are identical!");
+        }
+
+        int dx = abs(p1.getX() - p2.getX());
+        int dy = abs(p1.getY() - p2.getY());
+        util::Point curPoint = p1;
+
+        unsigned int fieldsToTraverse = dx + dy;
+        int incX = (p2.getX() > p1.getX()) ? 1 : -1;
+        int incY = (p2.getY() > p1.getY()) ? 1 : -1;
+        int error = dx - dy;
+        dx *= 2;                                            // scaling is needed to make sure the error term is integral
+        dy *= 2;
+
+        for (unsigned int fieldsVisited = 0; fieldsVisited < fieldsToTraverse; fieldsVisited++) {
+            if (curPoint == p2) {
+                break;
```

Comment:
Warum existiert dieser Case? Sollte der for-loop nicht abbrechen wenn man das letzte feld geprüft hat?